angular/projects/spark-angular/src/lib/components/sprk-autocomplete/sprk-autocomplete-result/sprk-autocomplete-result.directive.spec.ts

```diff
@@ -0,0 +1,119 @@
+import { Component } from '@angular/core';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { SprkAutocompleteResultDirective } from './sprk-autocomplete-result.directive';
+import { By } from '@angular/platform-browser';
+
+@Component({
+  selector: 'sprk-test',
+  template: `
+    <ul>
+      <li
+        sprkAutocompleteResult
+        analyticsString="result1"
+        idString="resultId1"
+        additionalClasses="testClass"
+      ></li>
+      <li sprkAutocompleteResult></li>
+    </ul>
+  `,
+})
+class TestComponent {}
+
+describe('Spark Autocomplete Result Directive', () => {
+  let component: TestComponent;
+  let fixture: ComponentFixture<TestComponent>;
+  let element: HTMLElement;
+  let directiveElement;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [SprkAutocompleteResultDirective, TestComponent],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(TestComponent);
+    component = fixture.componentInstance;
+    element = fixture.nativeElement.querySelector('li');
+
+    directiveElement = fixture.debugElement
+      .query(By.directive(SprkAutocompleteResultDirective))
+      .injector.get(
+        SprkAutocompleteResultDirective,
+      ) as SprkAutocompleteResultDirective;
+
+    fixture.detectChanges();
+  }));
+
+  it('should create itself', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should add the correct base class', () => {
+    expect(element.classList.contains('sprk-c-Autocomplete__result')).toEqual(
+      true,
+    );
+  });
+
+  it('should add classes if additionalClasses has a value', () => {
+    expect(element.classList.contains('testClass')).toEqual(true);
+  });
+
+  it('should add a value for data-analytics if analyticsString has a value', () => {
+    expect(element.getAttribute('data-analytics')).toEqual('result1');
+  });
+
+  it('should add a value for data-id if idString has a value', () => {
+    expect(element.getAttribute('data-id')).toEqual('resultId1');
+  });
+
+  it('should emit click event', (done) => {
+    let called = false;
+    directiveElement.clickedEvent.subscribe((g) => {
+      called = true;
+      done();
+    });
+    element.click();
+    expect(called).toEqual(true);
+  });
+
+  it('should add the class and aria attribute if isHighlighted is true', () => {
+    directiveElement.isHighlighted = true;
```

Comment:
The naming of `isHighlighted` speaks to a specific style vs. a function. With styles being fluid over time, naming this something more specific to the expected functionality would ensure we don't have breaking changes in the future. Maybe something like `isSelected` could work.